### ISSUE TITLE
[1.16.1] Head / Player Rotation gets reset when looking at a player.

### ISSUE THREAD

**Issue by Gbergz** [McJty/TheOneProbe#399]
Title^
Pretty self explanatory. I think this problem exist in 1.15.2 aswell.
This is re-producable in a clean instance with only Forge + TheOneProbe installed.

**Minecraft Ver:** 1.16.1
**Forge:** 32.0.77
**The One Probe Ver:** 3.0.1-beta

**GIF Showing the problem:**
https://gyazo.com/ccc935676cfff617d3d29a05403c221f

**-** Gbergz

**Comment by MrScautHD** [comment]
YES PLS FIX this otherwise is this a TRASH MOD!

**Comment by Gbergz** [comment]
> YES PLS FIX this otherwise is this a TRASH MOD!

Are you serious? No need to comment that. It'll be fixed eventually..

**Comment by kalkafox** [comment]
Reporting the same issue on a 1.15.2 modpack I am producing. This is also happening on 1.15.2. Here are the details to document this issue:

Similar to @Gbergz, I have a sample of the bug reproduced on the server [here.](https://imgur.com/KKVeamE) I will update with anything else relating to this rendering issue.

**Comment by McJty** [comment]
I haven't been able to find time for this yet but I will. It's also somewhat hard to simulate in a dev env

**Comment by kalkafox** [comment]
> 
> 
> I haven't been able to find time for this yet but I will. It's also somewhat hard to simulate in a dev env

I'm in the process of generating a gradle dev environment at the moment to see if I can trace it. I have a feeling it has something to do with the model rendering on the tooltip. I'll report back if I find anything.

EDIT: gradle setup is complete, off I go!

**Comment by McJty** [comment]
I'm pretty sure it has something to do with that. I also do some rotation client-side and its' likely that is messing things up

**Comment by kalkafox** [comment]
> 
> 
> I'm pretty sure it has something to do with that. I also do some rotation client-side and its' likely that is messing things up

Our suspicion was correct, it seems that setting static values for `rotationPitch` and `rotationYaw` on the entity property for `EntityPlayer` causes the bug, a crude solution is shown [here](https://imgur.com/fUvYypY)

Both `rotationPitch` and `rotationYaw`, beginning in Line 44 under the `RenderHelper` class are set to `0.0F`, which is where I first looked, and commented the `renderEntityStatic` function to see if it had anything to do with the rendering. I followed back to these values, which are locked at `0.0F` and seem to replicate the player's tooltip static facing model, which is what causes the player's position to reset (in layman's terms, it seems the rotation effect also applies to the player as it does on the tooltip, which can most likely be prevented by changing the property of the new entity rendered on the screen instead)

I literally don't know much about Java, so I could be wrong, but this could be a temporary solution to the problem. The only difference with this is that the player's head is not reset in the tooltip. Whether it is your intention to keep it reset is up to you, but I hope this helped!

A snippet of what I replaced from L44-L56:

```java
        entity.rotationPitch = entity.rotationPitch;
        entity.prevRotationPitch = entity.prevRotationPitch;
        entity.rotationYaw = entity.rotationYaw;
        entity.prevRotationYaw = entity.prevRotationYaw;

        if (entity instanceof LivingEntity) {
            LivingEntity livingEntity = (LivingEntity) entity;
            livingEntity.prevRenderYawOffset = livingEntity.prevRenderYawOffset;
            livingEntity.renderYawOffset = livingEntity.renderYawOffset;

            livingEntity.rotationYawHead = livingEntity.rotationYawHead;
            livingEntity.prevRotationYawHead = livingEntity.prevRotationYawHead;
        }
```

EDIT: I'm more than happy to set up a pull request for this issue, lemme know!
